crates/core_simd/src/array.rs

```diff
@@ -0,0 +1,165 @@
+use crate::intrinsics;
+use crate::masks::*;
+use crate::vector::ptr::SimdConst;
+use crate::vector::*;
+
+/// A representation of a vector as an "array" with indices,
+/// implementing operations applicable to any vector type and
+/// describing relationships between vector and scalar types.
+pub trait SimdArray<const LANES: usize>: crate::LanesAtMost32
```

Comment:
This trait should be sealed, I think, based on the function implementations.

Do we need to be concerned about backwards compatibility?  What happens if we want to add more functions to this trait in the future?

Reply:
How does anyone implement this for their own type when it requires them to implement LanesAtMost32?

Reply:
Hmm, that trait is sealed, so I guess it effectively is sealed already.

Reply:
Looking over https://rust-lang.github.io/rfcs/1105-api-evolution.html to be sure, it looks like we should be good with backwards compatibility as long as the trait is sealed.

Reply:
Yeah, I was thinking while I was writing this that we might want to move the actual sealing trait around, but that we can pick any sealed trait.